src/main/java/umk/mat/pajda/ProjektZespolowy/services/TipService.java

```diff
@@ -95,4 +101,26 @@ public ResponseEntity<String> createPayment(OpinionPostDTO opinionPostDTO, Strin
   public void setRestTemplate(RestTemplate restTemplate) {
     this.restTemplate = restTemplate;
   }
+
+  public TipStatisticsGetDTO getStatistics(String userName) {
+    try {
+      User user = userRepository.findByMail(userName).get();
+      TipStatisticsGetDTO tipStatisticsGetDTO = new TipStatisticsGetDTO();
+      Tip tipMin = tipRepository.findFirstByUserOrderByAmountDesc(user);
+      tipStatisticsGetDTO.setMaxTipAmount(tipMin.getAmount());
+      tipStatisticsGetDTO.setCurrency(tipMin.getCurrency());
```

Comment:
What if client will pay onece in PLN and then in USD

Reply:
resolved , implementation change